ds701_book/26-TimeSeries.qmd

```diff
@@ -0,0 +1,953 @@
+- $E_{t-s}, E_{t-2s}, \ldots, E_{t-Qs}$ are the seasonal lagged forecast errors.
+
+**Key Points**:
+
+1. **Order (p, d, q) x (P, D, Q, s)**: The parameters of the SARIMA model where $p, d, q$ are the non-seasonal parameters, $P, D, Q$ are the seasonal parameters, and $s$ is the length of the seasonal cycle.
+2. **Stationarity**: Differencing is used to make the series stationary.
+3. **Parameter Estimation**: Methods like Maximum Likelihood Estimation (MLE) are used to estimate the parameters.
+
+## ARIMA/SARIMA Example
+
+TODO: Add example of ARIMA/SARIMA using Python from other notebook.
+
+# Model Evaluation and Forecasting
+
+## Criteria for model selection (e.g., AIC, BIC)
+
+- **AIC (Akaike Information Criterion)**: A measure of the relative quality of a
+  statistical model for a given set of data. It is defined as:
+
+$$
+\text{AIC} = 2k - 2\ln(L)
+$$
+
+  where:
+
+  - $k$ is the number of parameters in the model,
+  - $L$ is the maximum value of the likelihood function for the model.
+
+**BIC (Bayesian Information Criterion)**: Similar to AIC but includes a penalty term for the number of parameters in the model. It is defined as:
+
+$$
+\text{BIC} = k\ln(n) - 2\ln(L)
+$$
+
+  where:
+
+  - $k$ is the number of parameters in the model,
+  - $n$ is the number of data points,
+  - $L$ is the maximum value of the likelihood function for the model.
+
+**Key Points**:
+
+  1. **Model Comparison**: Both AIC and BIC are used to compare different models; the model with the lower AIC or BIC is preferred.
+  2. **Penalty for Complexity**: BIC imposes a larger penalty for models with more parameters compared to AIC.
+  3. **Trade-off**: There is a trade-off between goodness of fit and model complexity.
+
+## Cross-validation techniques for time series
+
+- **Time Series Cross-Validation**: Unlike traditional cross-validation, time series cross-validation respects the temporal order of the data. Common techniques include:
+  - **Rolling Forecast Origin**: The training set is expanded with each iteration, and the model is re-evaluated.
+  - **Time Series Split**: The data is split into multiple training and test sets, ensuring that the training set always precedes the test set.
+  - **Blocked Cross-Validation**: The data is divided into blocks, and each block is used as a test set while the preceding blocks are used for training.
+
+## Key Points:
+1. **Respect Temporal Order**: Ensure that the training set always precedes the test set to avoid data leakage.
+2. **Multiple Techniques**: Various techniques like rolling forecast origin, time series split, and blocked cross-validation can be used.
+
+## Forecasting and confidence intervals
+
+- **Forecasting**: The process of making predictions about future values based on historical data. Common methods include:
+  - **ARIMA (AutoRegressive Integrated Moving Average)**: A popular time series forecasting method that combines autoregression, differencing, and moving average components.
```

Comment:
Is ARIMA the only model that is used for forecasting? Or do the other methods we introduced also work?

Reply:
Exponential smoothing can also be used, but seems the SARIMA family of models does better.